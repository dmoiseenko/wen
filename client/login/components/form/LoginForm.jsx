import React from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form';

import Input from '../../../common/components/input/Input';
import { emailField, passwordField } from '../../core/form.constants';


export default function LoginForm({ signIn }) {
  function handleClick(event) {
    event.preventDefault();
    signIn();
  }

  return (
    <form className="login-form">
      <Field
        name={emailField}
        component={Input}
        placeholder="mail@example.com"
        type="text"
        inputClassName="email-input"
        iconClassName="fa fa-user"
      />
      <Field
        name={passwordField}
        component={Input}
        placeholder="●●●●●●●"
        type="password"
        inputClassName="password-input"
        iconClassName="fa fa-lock"
      />
      <p className="control login">
        <button
          type="submit"
          onClick={handleClick}
          className="login-button"
        >
          Login
        </button>
      </p>
    </form>
  );
}

LoginForm.propTypes = {
  signIn: PropTypes.func.isRequired
};
