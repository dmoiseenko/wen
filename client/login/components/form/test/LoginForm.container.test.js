import * as container from '../LoginForm.container';


it('should map dispatch to actions correctly', () => {
  expect(container.mapDispatchToProps).toMatchSnapshot();
});

it('should setup from correctly', () => {
  expect(container.form).toMatchSnapshot();
});
