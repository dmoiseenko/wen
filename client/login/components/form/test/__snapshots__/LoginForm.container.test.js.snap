// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should map dispatch to actions correctly 1`] = `
Object {
  "signIn": [Function],
}
`;

exports[`should setup from correctly 1`] = `
Object {
  "form": "signInForm",
}
`;
